### ISSUE TITLE
stream line model: precision of angles 

### ISSUE THREAD

**Issue by jpinedaf** [jpinedaf/velocity_tools#3]
In one of the tests that @tere-valdivia  carried out, it was identified that the precision of the angles in the streamline model was insufficient for some very specific cases. 
The angle theta is defined in an implicit equation, and the angle is then calculated numerically using `scipy.optimize.minimize`

An initial patch is provided in #2 where a `tol` parameter is used to set the desired tolerance required in the `minimize` function.

I think that possibly a more robust way to define this value. Maybe we could try to use the ratio between radial and angular velocity to scale the tolerance to some value, maybe something like this
```python
tol = C * delta radius * Omega / (v_radial + epsilon) 
```
where epsilon is a value to make sure that things don't diverge when using small (or zero) radial velocity, and `C` is a constant that is calibrated to work in the example model. What do you think?

**Comment by tere-valdivia** [comment]
We brainstormed about this and I will test the following:

Instead of making the tolerance a free parameter, we will test placing the spacing between consecutive values of the radius r. then delta r in the formula above is the free parameter. I will test values of epsilon and C to see what is best.

Also, I found out that the initial guess for the theta angle also affects the `scipy.optimize.minimize`, so I will see if it is ok to use an initial guess of theta0 + some factor of the tolerance.

I'll keep you posted.

**Comment by tere-valdivia** [comment]
Hi,

I tested different combinations of C and v and while doing that, I realized that when we gave the parameter `tol`to the `optimize.minimize`, I did not really know to which parameter it gave it afterwards, as depending on the method selected by the minimization, there are different tolerances and epsilons to look for.

I left C=6e-5. and epsilon = 0.1 km/s

I investigated and by default, when `optimize.minimize` receives bounds and no constrains, it uses the L-BFGS-B method. It has two tolerances and an epsilon. What each means can be found in the [scipy entry.](https://docs.scipy.org/doc/scipy/reference/optimize.minimize-lbfgsb.html#optimize-minimize-lbfgsb)

To have control over the tolerances I pass to the function `optimize.minimize`, I passed them as a dictonary:
```python
options_dict = {'gtol': tol/10., 'eps': tol, 'ftol': tol}
```
through the `options` parameter. 

For now, this gives a radial velocity different to zero after the first point down to `Omega0 = 6e-15` and `v_r0 = 0`. I think that is quite a low angular velocity for most cases.
